### PR TITLE
ref(widget-builder): Implement widget new design part 4

### PR DESCRIPTION
**Pull request by priscilawebdev**
. Separates Widget Builder in two widgets: Events and Metrics
. Adds dataSet URL parameter
. Introduces the component EditableText, providing a way to edit the widget title 
. Uses sessions API to display metric data

Preview:

image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/pickProjectToContinue.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import ContextPickerModal from 'app/components/contextPickerModal';
+import {Organization, Project} from 'app/types';
+
+type Props = {
+  orgSlug: Organization['slug'];
+  router: ReactRouter.InjectedRouter;
+  projects: Project[];
+};
+
+const PickProjectToContinue = ({orgSlug, router, projects}: Props) => {
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/pickProjectToContinue.tsx:15]: This seems to be a duplication of the one in release detail - should we consolidate?

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/queries.tsx [hunk 2]
```diff
@@ -1,65 +1,104 @@
...
       const newQuery = {...widgetQuery, [field]: value};
       onChangeQuery(queryIndex, newQuery);
     };
   }
 
-  function getTagValue({key}: Tag, _query: string, _params: object) {
-    return Promise.resolve([metricTags[key as keyof typeof metricTags].value]);
-  }
+  // // this is just a mock
+  // function fetchTagValues(tagKey: string) {
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/queries.tsx:48]: Should we remove these comments (and import ones) before merging?

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/statsRequest.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+          interval: getInterval(datetime),
+          ...getParams(pick(location.query, Object.values(URL_PARAM))),
+        },
+      });
+    });
+
+    let completed = 0;
+    const sessionReponses: SessionApiResponse[] = [];
+
+    promises.forEach(async promise => {
```
Comment (matejminar) [inline src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/statsRequest.tsx:97]: Are we not using Promise.all on purpose here?

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/pickProjectToContinue.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/pickProjectToContinue.tsx:15]: sure... I am working a consolidation in this PR https://github.com/getsentry/sentry/pull/25183

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/queries.tsx (hunk 2 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/queries.tsx:48]: sure thing!

File: src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/statsRequest.tsx (hunk 3 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/dashboardsV2/widget/metricWidget/statsRequest.tsx]: updated 😉

File: src/sentry/static/sentry/app/components/pickProjectToContinue.tsx [hunk 4]
```diff
@@ -31,7 +31,7 @@ function PickProjectToContinue({
 }: Props) {
   let navigating = false;
 
-  const path = `${nextPath}?project=`;
+  const path = nextPath.includes('?') ? `${nextPath}&project=` : `${nextPath}?project=`;
```
Comment (matejminar) [inline src/sentry/static/sentry/app/components/pickProjectToContinue.tsx:34]: Can there be other question marks in the path other than the one you mean?

**Review by matejminar — APPROVED** [review]

File: src/sentry/static/sentry/app/components/pickProjectToContinue.tsx (hunk 4 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/pickProjectToContinue.tsx:34]: not at the moment, but in the future, if necessary, we can update this logic 😉

**Pull request merged**